src/main/java/pl/edu/mimuw/collections/MultiSet.java

```diff
@@ -0,0 +1,126 @@
+package pl.edu.mimuw.collections;
+
+import java.util.*;
+
+public class MultiSet<E> implements IMultiset<E>, Iterable<E> {
+
+  private static final int MAX_VISIBLE_SIZE = 10;
+
+  private final Map<E, Integer> elements;
+
+  public MultiSet() {
+    this.elements = new HashMap<>();
+  }
+
+  @Override
+  public int size() {
+    int result = 0;
+    for (var v : this.elements.values()) {
+      result += v;
+    }
+    return result;
+  }
+
+  @Override
+  public int count(E element) {
+    if (this.elements.get(element) == null) return 0;
+    return this.elements.get(element);
+  }
+
+  @Override
+  public int add(E element, int occurrences) {
+    if (this.elements.get(element) == null) {
+      this.elements.put(element, occurrences);
+      return occurrences;
+    }
+    int tmp = this.elements.get(element);
+    this.elements.remove(element);
+    this.elements.put(element, tmp + occurrences);
+    return tmp + occurrences;
+  }
+
+  @Override
+  public boolean add(E element) {
+    if (this.elements.get(element) == null) {
+      this.elements.put(element, 0);
+    }
+    int tmp = this.elements.get(element);
+    this.elements.remove(element);
+    this.elements.put(element, tmp + 1);
+    return true;
+  }
+
+  @Override
+  public int remove(E element, int occurrences) {
+    int tmp = this.elements.get(element);
+    this.elements.remove(element);
+    if (tmp > occurrences) this.elements.put(element, tmp - occurrences);
+    return tmp - occurrences;
+  }
+
+  @Override
+  public boolean remove(E element) {
+    int tmp = this.elements.get(element);
+    this.elements.remove(element);
+    if (tmp > 1) this.elements.put(element, tmp - 1);
+    return true;
+  }
+
+  @Override
+  public Set<E> elementSet() {
+    Set<E> result = new HashSet<>();
+    for (var v : this.elements.keySet()) {
+      for (int i = 0; i < this.elements.get(v); i++) {
```

Comment:
ten `for` jest niepotrzebny, `Set` nie jest multisetem i chodziło o stworzenie seta z tylko jedna kopia kazdego elementu (zadziala dobrze, ale nieefektywnie)